src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritItemListenerTest.java

```diff
@@ -92,4 +101,38 @@ public void testOnLoadedWithNoConnection() throws Exception {
         listener.onLoaded();
         Mockito.verify(gerritServer, Mockito.times(0)).startConnection();
     }
+
+    /**
+     * Test {@link GerritItemListener#onLocationChanged(hudson.model.Item, String, String)} do handle renamed job.
+     *
+     * @throws Exception if so.
+     */
+    @Test
+    @Issue("JENKINS-27651")
+    public void testOnJobRenamed() throws Exception {
+        FreeStyleProject job = j.createFreeStyleProject("MyJob");
+        PatchsetCreated event = Setup.createManualPatchsetCreated();
+
+        GerritTrigger trigger = spy(new GerritTrigger(
+                null, null, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                true, true, true, false, false, "", "", "", "", "", "", "", null, null, null,
+                null, false, false, "", null));
+
+        doReturn(new GerritTrigger.DescriptorImpl()).when(trigger, "getDescriptor");
+        job.addTrigger(trigger);
+        trigger.start(job, true);
+
+        when(trigger.isInteresting(event)).thenReturn(true);
+        job.renameTo("MyJobRenamed");
+        GerritItemListener listener = new GerritItemListener();
```

Comment:
Since the JenkinsRule is part of the test I doubt you need to create a new listener and call ```onLocationChanged``` on it, ```job.renameTo``` should do that, rught?